pyam/str.py

```diff
@@ -4,6 +4,8 @@
 import pandas as pd
 from pandas.api.types import is_list_like
 
+REGEXP_CHARACTERS = r".+()[]{}|$"
```

Comment:
Not that I'd ever seen it but we can add the caret symbol as well `^`, especially since you have the dollar sign:
```suggestion
REGEXP_CHARACTERS = r".+()[]{}|$^"
```

Reply:
Thanks for this comment - went through the list of regex-characters at https://docs.python.org/3/library/re.html and also added `?`...